src/App.tsx

```diff
@@ -17,18 +53,40 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                todos={todos}
+                setPreparedTodos={setPreparedTodos}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && (
+                <Loader />
+              )}
+
+              {isLoadingFailed && (
+                <p>No todos were loaded!</p>
+              )}
+
+              {!isLoading && !isLoadingFailed && (
```

Comment:
seems quite wordy for a positive result 😄 
I'd say:
```suggestion
              {areTodosReady && (
```